src/content/docs/en/guides/server-islands.mdx

```diff
@@ -33,6 +33,15 @@ const avatarURL = await getUserAvatar(userSession);
 ---
 <img alt="User avatar" src={avatarURL} />
 ```
+:::caution[Server Island Prop Limitations]
+You can pass a function as a prop to a server island component, but this only works during server rendering. If you try to use the function in a hydrated component (for example, as an event handler), an error will occur.
```

Comment:
> If you try to use the function in a hydrated component (for example, as an event handler), an error will occur.

This example seems weird to me. Server islands are not hydrated (at least not in the sense we use anywhere else in the docs for that word, it doesn't add handlers and bindings for interactivity to existing DOM nodes). They get replaced with the content of a separate request when loaded.

The fallback/original content doesn't require any hydration and neither does the code after the swap. After the swap there isn't even any JS related to the server island in the document, contrary to hydration, which is also a win for server island, and might give people the idea that server islands are heavier on their site than it really is.